docs/user-guide.md

```diff
@@ -341,6 +343,7 @@ won't need to know.
 
 There are a couple of places where the syntax of type annotations like `@Nullable`
 is counterintuitive.
+TODO: make the principle clear: type use annotation immediately before the annotated type.
```

Comment:
I'm not sure I agree with the `counterintuitive` part... we need to make the principle clear: a type use annotation is immediately before/to the left of the annotated type. Then the two examples should be clear.

Reply:
That's a good point for us to emphasize.

The counterintuitiveness was there already, because `String[]` (array of string) is backwards from `List<String>` (list of string), and because in `Map.Entry`, despite completely resembling `OuterType.InnerType`, `Map` is only a name qualifier and not a type usage (and only checking for `static` at the nested declaration would tell you).

Reply:
I have to say that it's not clear to me that "immediately to the left of the annotated type" helps explain the placement of `@Nullable` in `String @Nullable []`. `[]` isn't a type. Things get even worse with multi-dimensional arrays. I'm not sure there's a simple principle to state here, and I think just showing examples remains the way to go.